internal/clients/nightfall/nightfall.go

```diff
@@ -262,19 +286,52 @@ func (n *Client) scanAllContent(ctx context.Context, logger logger.Logger, cts [
 }
 
 // Scan send /scan request to Nightfall API and return findings
-func (n *Client) Scan(ctx context.Context, logger logger.Logger, items []string) ([][]nightfallAPI.ScanResponse, error) {
+func (n *Client) Scan(
+	ctx context.Context,
+	logger logger.Logger,
+	items []string,
+) ([][]nightfallAPI.ScanResponse, error) {
 	APIKey := nightfallAPI.APIKey{
 		Key:    n.APIKey,
 		Prefix: "",
 	}
 	newCtx := context.WithValue(ctx, nightfallAPI.ContextAPIKey, APIKey)
-	request := n.createScanRequest(items)
-	resp, _, err := n.APIClient.ScanAPI().ScanPayload(newCtx, request)
-	if err != nil {
-		logger.Error(fmt.Sprintf("Error from Nightfall API, unable to successfully scan %d items", len(items)))
-		return nil, err
+	request := n.CreateScanRequest(items)
+	return n.makeScanRequestWithRetries(newCtx, logger, request)
+}
+
+func (n *Client) makeScanRequestWithRetries(
+	ctx context.Context,
+	logger logger.Logger,
+	request nightfallAPI.ScanRequest,
+) ([][]nightfallAPI.ScanResponse, error) {
+	delay := n.InitialRetryDelay
+	for i := 0; i < maxRetries; i++ {
+		resp, httpResp, err := n.APIClient.ScanPayload(ctx, request)
+		if err != nil {
+			if httpResp != nil && httpResp.StatusCode == http.StatusTooManyRequests {
```

Comment:
can this httpResp be nil?

Reply:
yup there's an error case that returns a nil httpResp